cmd/deploy.go

```diff
@@ -293,3 +289,38 @@ func setFlags(conf *clab.Config) {
 		conf.Mgmt.IPv6Subnet = v6
 	}
 }
+
+// determineWorkerCount calculates the number of parralel threads for the creation of links and nodes.
+// The maximum number of Threads reported for links and nodes is the definet number of nodes and links provided.
+// If maxWorkerCount is > 0, it will be used as a cut-off value. At max the maxWorkerCount threads will be returned
+// However if the node or link count is less, just the node or link count is returned.
+// If maxWorkerCount is unset (<=0) then the number of vCPUs is considered as the cut-off value also considering the
+// amount of links and nodes. returned nodeWorker and linkWorker counts will never be more then the provided nodeCount and linkCount.
+func determineWorkerCount(nodeCount, linkCount, maxWorkerCount uint) (nodeWorkers, linkWorkers uint, err error) {
+	// init number of Workers to the number of links and nodes
+	nodeWorkers = nodeCount
+	linkWorkers = linkCount
+
+	// default maxThreadCount to maxWorkerCount
+	maxThreadCount := maxWorkerCount
+	// if maxWorkerCount is not set (==0) use cpu count as max
+	if maxWorkerCount <= 0 {
+		// retrieve vCPU count
+		vCpus, err := cpu.Counts(true)
```

Comment:
hi @steiler 
can you use this package instead https://github.com/tklauser/numcpus

easier on dependencies

Reply:
Yes, done!